---
#title: Predicting heart rejection response with spatial transcriptomics
talk-title: "Multimodal Integration for Heart Transplant Rejection"
author: Chunqing (Tony) Liang
other-authors: "Dr. Amrit Singh"
institute: University of British Columbia
talk-date: "Oct 6, 2025"
---

## {#title background-image="assets/img/website_no_bkg.png" background-position="top"}

<!--- setup r lib -->
```{r}
#| label: "setup"
#| echo: false
#| include: false
#| eval: true

library(knitr)
library(cowplot)
```

<br>
<br>
<br>
<br>
<br>


### {{< meta talk-title >}} 

<br>


::: {.f3 .bold}

{{< meta author >}} 

PhD student in Bioinformatics

Supervisor: {{< meta other-authors >}}

{{< meta talk-date >}}

:::

## Land acknowledgement {background-image=assets/img/land_ack.jpg background-size=cover background-opacity="0.4"}

**I would like to acknowledge that I work on the traditional, ancestral, and unceded territory of the Coast Salish Peoples, including the territories of the xwməθkwəy̓əm (Musqueam), Skwxwú7mesh (Squamish), Stó:lō and Səl̓ílwətaʔ/Selilwitulh (Tsleil- Waututh) Nations.**

*Traditional*: Traditionally used and/or occupied by Musqueam people

*Ancestral*: Recognizes land that is handed down from generation to generation

*Unceded*: Refers to land that was not turned over to the Crown (government) by a treaty or other agreement

::: {.notes}

* Be short on this slide
* Traditional, ancestral, unceded territory of Coast Salish people, inckuding territories of :
  * Musqueam
  * Squamish
  * Tsleil-Waututh Nations

:::

## Heart Transplant

- Replace person'sick heart with healthy one from someone who has died

- Occurs when heart failure is end-stage and no longer treatable in any other way

## Heart Transplant Rejection

:::: {.columns}

::: {.column}

```{r}
#| fig-align: center
#| fig-cap-location: bottom
#| fig-cap: "Types of heart transplant rejection [@lipkova2022deep]"

knitr::include_graphics("assets/img/heart_transplant_rejection.png")
```

:::

::: {.column}

- Relies on endomyocardial biopsy (EMB)
- Assessments from pathologists, subjective and opinion-based
- Multiple rejection types
  - [Acute cellular rejection]{.secondary} (ACR)
  - Antibody-mediated rejection
  - [Quilty lesion]{.tertiary}
  - Look similary structurally

:::

::::

::: {.notes}

- Diagnosis relies on endomyocardial biopsies graded by pathologists 

:::

## Quilty and ACR


```{r}
#| fig-align: center
#| fig-cap-location: bottom
#| fig-cap: "Quilty lesion and acute cellular rejection. (A) Quilty lesion. (B) Grade 1R. (C) Grade 2R. (D) Grade 3R. [@cho2019quilty]"
knitr::include_graphics("assets/img/acr_and_quilty.jpg")
```



## How to improve diagnosis?

- What's better way to assist pathologists to differentiate different rejections
  - What to do if there's disagreement?

- Computational pathology comes in handy!

::: {.fragment}

```{r}
#| fig-align: center
#| fig-cap-location: bottom
#| fig-cap: "Workflow in computational pathology [@wagner2022make]"

knitr::include_graphics("assets/img/comp_pathology_flow.png")
```

:::


## In-house data

:::: {.columns}

::: {.column}

```{r}
#| fig-align: center
#| fig-cap-location: bottom

knitr::include_graphics("assets/img/cohort_timeline.png")
```

:::

::: {.column}

- Data from: 
  - Prevention of Organ Failure ([PROOF]{.tertiary}) centre
  - Bruce McManus Cardiovascular Biobank ([BMCB]{.tertiary})
- Cohort of $> 3000$ samples from $500$ patients
- Composed of different types of no rejection, ACR, AMR, mixed rejection

:::

::::

## Available variable in the cohort

```{r}
#| eval: true
#| echo: false

library(knitr)
library(kableExtra)

# Create a data frame
data_dict <- data.frame(
  Category = c(
    "Demographics",
    "Participant and donor viral serology",
    "Induction therapy",
    "Vitals",
    "Labs",
    "Complete blood counts with differentials",
    "Immunosuppressant blood levels",
    "Human Leukocyte Antigen – Panel Reactive Antibody (HLA-PRA)",
    "Medication",
    "EMB biopsy",
    "Graft outcomes"
  ),
  Variables = c(
    "Age, sex, ethnicity, comorbidities",
    "Positive/negative/not done: Epstein-Barr Virus (EBV), Cytomegalovirus (CMV), HIV, HCV, HBsAg, HBsAb, HBcAg, HBcAb, HBV DNA",
    "Medication with dose, route and timelines",
    "Smoking status, weight, height, temperature, heart rate, systolic & diastolic blood pressure",
    "Glucose, creatinine, eGFR, urea, sodium, potassium, chloride, bicarbonate, calcium, CRP, BNP, NT-proBNP, hemoglobin, platelet",
    "Hemoglobin, platelets, WBC, RBC, neutrophils, hematocrit, lymphocytes, monocytes, eosinophils, basophils, MCV, RDW, MPV",
    "Time and dose of tacrolimus, cyclosporine, sirolimus",
    "Participant and donor HLA alleles: A1, A2, B, C, DR, DQ, DP",
    "Medication times, dosing, route for both maintenance immunosuppression, hypertension, and statin therapy",
    "ISHLT grade, antibody mediated rejection, Quilty lesions, late ischemic injury",
    "Echocardiographic data (ejection fraction, diastolic function (E/A, E/e’), chamber volumes)"
  )
)

# Render table
DT::datatable(
  data_dict,
  rownames = FALSE,
  options = list(
    pageLength = 3,     # show 10 rows per page
    autoWidth = TRUE,    # auto adjust column widths
    scrollX = TRUE       # allow horizontal scrolling
  )
)
```


## Proof Centre

::: {.people-row}
![](assets/img/proof/PROOF_logo_cond_RGB.png){.logo}
![](assets/img/proof/dr_tebutt.jpg){.portrait}
![](assets/img/proof/mr_casey_shannon.jpg){.portrait}
![](assets/img/proof/ms_sarah_assadian.jpg){.portrait}
:::

- Lead by Dr. Scott Tebutt, Mr. Casey Shannon, Ms. Sarah Assadian
- Majority of biopsies of the cohort is from here
- Collected data from multiple participating sites like Vancouver, Ottawa and Nebraska
- Multiple timepoints biopsies from patients to from longitudinal study

## Bruce McManus Cardiovascular Biobank


:::: {.people-row}

::: {.logo-container}
![](assets/img/bruce_biobank/bruce_mcmanus_logo.jpg){.round-logo}
:::
![Dr. Ying Wang](assets/img/bruce_biobank/Ying-Wangnew.jpg){.round}
![Dr. Chi Lai](assets/img/bruce_biobank/chi_lai.jpeg){.round}
![](assets/img/bruce_biobank/gurpreet_singhera.jpg){.round}
![](assets/img/bruce_biobank/coco_ng.jpeg){.round}
![](assets/img/bruce_biobank/tiffany_chang.jpg){.round}
::::

<br>

- Lead by Dr. Ying Wang, Dr. Chi Lai, Dr. Singhera, Ms. Coco Ng, Ms. Tiffany Chang
- Several hundred more local heart transplant patients from here
- Currently restoring slides from storage and digitizing them


## Objective

- Improve diagnosis using multimodal data

- Can we tell long term outcome of rejection?
  - i.e. Cardiac allograft vasculopathy ([CAV]{.secondary})
  - CAV is 1 of top 3 causes of death in the first year after heart transplant and increases in prevalence over time [@shetty2020heart]

::: {.notes}

- CAV progressive condition that affects the arteries, arterioles and capillaries post heart transplant

:::
## Improve diagnosis using mulitmodal data

```{r}
#| fig-align: center
#| fig-cap-location: bottom

knitr::include_graphics("assets/img/aim1_develop_model_classification.png")
```


## Can we tell long term outcome of rejection


::: {.bottom-align}
![](assets/img/pca_cav.png)
![](assets/img/pca_cav2.png)
:::


- Rejection have $\geq 1R$ is more likely to have CAV
- People that have CAV have similar characteristics with others


## Types of available methods

::: {.columns}

::: {.column width="45%"}

```{r out.width="600px", out.height="480px"}
#| fig-align: center
#| fig-cap-location: bottom
#| fig-cap: "Types of multimodal integration method [@subramanian2020multi]"
knitr::include_graphics("assets/img/methods_avail.jpg")
```

:::
<!--- Empty column here --->
::: {.column width="5%"} 
<!--- Empty column here --->
:::


::: {.column width="50%"}

<br>

::: {.callout-note .fragment}

# Question arises

- Many integration methods 
  
  ---> Which to use, how to choose them?

- [Reproducibility crisis]{.secondary}
  
  ---> How to reproduce method and get [reliable]{.tertiary} results?

- Existing benchmark studies are not 100% complete or all-encompassing

  ---> Technical difficulty in implementation?

:::

:::

::::

::: {.notes}

- Earlier talk about data, but which way we know could maximize / make use of full potential from data
- Then, which task are we performing, and how do you know if it is better in your data not just by chance?
- Fast way to validate them? Hard to reproduce, not really work on custom data
- Have anyone try setting this validation up? 
  - Reliable? Covered all? Easy to access? 

:::

## MESSI pipeline

- We created a workflow pipeline [*Multimodal Experiments with SyStematic Interrogation using nextflow*]{.secondary}
- Solves reproducibility issue through:
  - [Nextflow]{.primary} [@di2017nextflow], workflow managment
  - Singularity [@kurtzer2017singularity], software environment control
- Looking for collaborations


## MESSI pipeline

![](assets/img/overview_fig/MESSI_workflow_overview_diagram.png){fig-align="center"}


## Standardized data preprocessing

![](assets/img/overview_fig/overview_upper_panel.png){fig-align="center"}


- Currently handles both R and Python --> could extend to more
- Data flows in there , handles N different datasets
- Important model selection prior to evaluation


## Flexible method evaluation

:::: {.columns}

::: {.column width="55%"}

```{r}

#| fig-align: center
#| fig-cap-location: bottom

knitr::include_graphics("assets/img/overview_fig/overview_mid_panel.png")
```


<!-- ![](assets/img/overview_fig/overview_mid_panel.png)
{fig-align="center"} 
#| label: "fig-overview-mid-panel"
#| fig-cap: "MESSI Method evaluation"
-->


:::

::: {.column width="45%"}

- Each method is an isolated workflow (dash box)
- Each method internally runs evaluation in parallel
- Each result is saved separately ...
- Easily extended to other tasks not just CV
- On / Off to run interested methods only

:::

::::

## Summarized reports of metrics performances

![](assets/img/overview_fig/overview_bottom_panel.png){fig-align="center" width="80%"}


- "Recursively" collects output from each method including versions
- Summarizes metrics for downstream analysis
- Provides rich report of computational resource usages


## Datasets evaluated {.smaller}

```{r}
#| eval: true
#| echo: false

dataset_table <- read.csv("data/real_dataset_metadata.csv")

DT::datatable(
  dataset_table,
  rownames = FALSE,
  options = list(
    pageLength = 5,     # show 5 rows per page
    autoWidth = TRUE,    # auto adjust column widths
    scrollX = TRUE       # allow horizontal scrolling
  )
)
```

## Computational Resources Usage {.nostretch}

:::: {.columns}

::: {.column width="60%"}

```{r out.width="600px", out.height="480px"}
#| fig-align: center
#| fig-cap-location: bottom
knitr::include_graphics("assets/img/fig_computational_resources_usage.png")
```

:::

::: {.column width="40%"}

- MOGONET and multiview require longer durations compared to rest
- MOGONET takes more virtual memory due to its deep learning nature
  - Transferring of data from GPU to CPU
- Model training and prediction steps demand more memory consumption

:::

:::: 



## Performance on real-world datasets

:::: {.columns}

::: {.column width="60%"}

```{r out.width="600px", out.height="480px"}
#| fig-align: center
#| fig-cap-location: bottom
knitr::include_graphics("assets/img/fig_performance_evaluation_real.png")
```

:::

::: {.column width="40%"}

- No method perfrom well in all scenarios
- DIABLO performs at top compared to others
  - The null design variant beats the full design
- MOGONET performs the weakest


:::

:::: 

## Biological Interpretation on biomarkers identified

![](assets/img/fig_fgsea_analysis_two_panels.png)

## MESSI future direction {.smaller}

- "**No method works universally well on all datasets**"
- Classic statistical methods still work, and sometimes even better than deep learning (DL)
- Pipeline proves way to [reproducibly]{.tertiary} explore, benchmark different aspects of integration methods
  - Resumable
  - Parallel to compute as many resources as allowed at the same time
  - Ease burden of [setting up environment]{.secondary}

- Need to add more methods and datasets
  - Especially DL models are more popular now
  - Explore if any dataset could have relation to another despite different disease/condition



## Current work on GeoMX

- TODO

## Background of GeoMX

- TODO

## GeoMxTools PCA

![](assets/img/geomx_results/class_segment_pca_spe_plot.png)

## GeoMxTools Volcano plot

![](assets/img/geomx_results/acr_vs_quilty_volcano_plot.png)

## GeoMxTools Significant genes heatmap


![](assets/img/geomx_results/acr_quilty_sig_gene_heatmap.png)



## Next steps

- TODO

## Conclusion

- TODO

## Discussion & Future directions {.smaller}

- "**No method works universally well on all datasets**"
- Classic statistical methods still work, and sometimes even better than deep learning (DL)
- Pipeline proves way to [reproducibly]{.tertiary} explore, benchmark different aspects of integration methods
  - Resumable
  - Parallel to compute as many resources as allowed at the same time
  - Ease burden of [setting up environment]{.secondary}

- Need to add more methods and datasets
  - Especially DL models are more popular now
  - Explore if any dataset could have relation to another despite different disease/condition


# Thanks! {.thanks}

## {#acknowledgements .smaller}

### Acknowledgements

- [Dr. Amrit Singh]{.bold}
- Dr. Maryam Ahmadzadeh
- Dr. Young Woong Kim
- Rishika Daswani
- Roy He
- Samuel Leung
- Raam Sivakumar
- Jeffrey Tang
- Michael Yoon
- Mingming Zhang


<!-- ::: {.footer}
![](assets/img/website_no_bkg.png)
::: -->


## Reference

::: {#refs}
:::